core/src/test/java/org/infinispan/util/DeadlockDetectingLockManagerTest.java

```diff
@@ -31,77 +30,72 @@
 @Test(groups = "unit", testName = "util.DeadlockDetectingLockManagerTest")
 public class DeadlockDetectingLockManagerTest extends AbstractInfinispanTest {
 
-   DeadlockDetectingLockManagerMock lockManager;
-   Configuration config = new ConfigurationBuilder().build();
-   private LockContainer lc;
-   private static final int SPIN_DURATION = 1000;
-   private DldGlobalTransaction lockOwner;
+   private static final long SPIN_DURATION = 1000;
+   private DeadlockDetectingLockManagerMock lockManager;
 
    @BeforeMethod
    public void setUp() {
-      lc = mock(LockContainer.class);
-      lockManager = new DeadlockDetectingLockManagerMock(SPIN_DURATION, true, lc, config);
-      lockManager.injectTimeService(TIME_SERVICE);
-      lockOwner = (DldGlobalTransaction) TransactionFactory.TxFactoryEnum.DLD_NORECOVERY_XA.newGlobalTransaction();
+      lockManager = new DeadlockDetectingLockManagerMock(createLockContainer(), createConfiguration());
+      lockManager.init();
    }
 
-
    public void testNoTransaction() throws Exception {
-      InvocationContext nonTx = new NonTxInvocationContext(null, AnyEquivalence.getInstance());
-
-      Lock mockLock = mock(Lock.class);
-      when(lc.acquireLock(nonTx.getLockOwner(), "k", config.locking().lockAcquisitionTimeout(), TimeUnit.MILLISECONDS)).thenReturn(mockLock).thenReturn(null);
-
-      assert lockManager.lockAndRecord("k", nonTx, config.locking().lockAcquisitionTimeout());
-      assert !lockManager.lockAndRecord("k", nonTx, config.locking().lockAcquisitionTimeout());
-
+      lockManager.lock("k", "aThread", 0, TimeUnit.MILLISECONDS).lock();
+      try {
+         lockManager.lock("k", "anotherThread", 0, TimeUnit.MILLISECONDS).lock();
+         fail("TimeoutException expected!");
+      } catch (TimeoutException e) {
+         //expected
+      }
    }
 
    public void testLockHeldByThread() throws Exception {
-      InvocationContext localTxContext = buildLocalTxIc(new DldGlobalTransaction());
-
-      Lock mockLock = mock(Lock.class);
-      //this makes sure that we cannot acquire lock from the first try
-      when(lc.acquireLock(localTxContext.getLockOwner(), "k", SPIN_DURATION, TimeUnit.MILLISECONDS)).thenReturn(null).thenReturn(mockLock);
-      lockManager.setOwner(Thread.currentThread() );
-      //next lock acquisition will succeed
+      lockManager.lock("k", "aThread", 0, TimeUnit.MILLISECONDS).lock();
+      try {
+         lockManager.lock("k", new DldGlobalTransaction(), 1000, TimeUnit.MILLISECONDS).lock();
+         fail("TimeoutException expected!");
+      } catch (TimeoutException e) {
+         //expected
+      }
 
-      assert lockManager.lockAndRecord("k", localTxContext, config.locking().lockAcquisitionTimeout());
-      assert lockManager.getOverlapWithNotDeadlockAwareLockOwners() == 1;
+      AssertJUnit.assertTrue(lockManager.getOverlapWithNotDeadlockAwareLockOwners() > 0); //it's not deterministic. can be 1 or more.
```

Comment:
well, the non deterministic in the test is caused because the deadlock check is done asynchronously in the schedule executor. 

if you really want, I can remove the attribute and the test.